plugins/org.jkiss.dbeaver.model/src/org/jkiss/dbeaver/model/impl/app/DefaultCertificateStorage.java

```diff
@@ -138,14 +135,21 @@ public void addCertificate(@NotNull DBPDataSourceContainer dataSource, @NotNull
             CertificateFactory cf = CertificateFactory.getInstance("X.509");
             List<Certificate> certChain = new ArrayList<>();
             if (caCertData != null) {
-                Certificate caCert = cf.generateCertificate(new ByteArrayInputStream(caCertData));
-                keyStore.setCertificateEntry(CA_CERT_ALIAS, caCert);
+                List<? extends Certificate> certificates =
+                    new ArrayList<>(cf.generateCertificates(new ByteArrayInputStream(caCertData)));
+                for (int i = 0; i < certificates.size(); i++) {
+                    keyStore.setCertificateEntry(i == 0 ? CA_CERT_ALIAS : CA_CERT_ALIAS + i, certificates.get(i));
```

Comment:
Let's extract `i == 0 ? CA_CERT_ALIAS : CA_CERT_ALIAS + i` to a separate utility method so it's not duplicated.